apps/mux/src/config.tsx

```diff
@@ -0,0 +1,295 @@
+    this.space = this.props.sdk.space;
+
+    // `onConfigure` allows to configure a callback to be
+    // invoked when a user attempts to install the app or update
+    // its configuration.
+    this.app.onConfigure(() => this.onConfigure());
+  }
+
+  async componentDidMount() {
+    // Get current parameters of the app.
+    const [parameters, eisRes, contentTypesRes] = await Promise.all([
+      this.app.getParameters(),
+      this.space.getEditorInterfaces(),
+      this.space.getContentTypes(),
+    ]);
+
+    const { ids } = this.props.sdk;
+
+    const compatibleFields = getCompatibleFields(
+      contentTypesRes.items as ContentType[]
+    );
+    const selectedFields = editorInterfacesToSelectedFields(
+      eisRes.items as EditorInterface[],
+      ids.app
+    );
+
+    this.setState(
+      // If the app is not installed, `parameters` will be `null`.
+      // We default to an empty object in this case.
+      {
+        parameters: parameters || {},
+        compatibleFields,
+        contentTypes: contentTypesRes.items as ContentType[],
+        selectedFields,
+      },
+      () => {
+        // Once preparation has finished, call `setReady` to hide
+        // the loading screen and present the app to a user.
+        this.app.setReady();
+      }
+    );
+  }
+
+  assignToField(contentTypeId: string, fieldId: string, enabled: boolean) {
+    const { selectedFields } = this.state;
+    selectedFields[contentTypeId] = [...(selectedFields[contentTypeId] || [])];
+    if (enabled) {
+      selectedFields[contentTypeId]!.push(fieldId);
+    } else {
+      selectedFields[contentTypeId] = selectedFields[contentTypeId]!.filter(
+        (id) => id !== fieldId
+      );
+    }
+    if (!selectedFields[contentTypeId]!.length)
+      delete selectedFields[contentTypeId];
+    this.setState({ selectedFields: { ...selectedFields } });
+  }
+
+  isChecked(contentTypeId: string, fieldId: string) {
+    return (
+      this.state.selectedFields[contentTypeId] &&
+      this.state.selectedFields[contentTypeId]!.includes(fieldId)
+    );
+  }
+
+  // Renders the UI of the app.
+  render() {
+    const {
+      parameters: { muxAccessTokenId, muxAccessTokenSecret },
+      contentTypes,
+      compatibleFields,
+    } = this.state;
+
+    return (
+      <React.Fragment>
+        <div className="config-background" />
+        <div className="config-body">
+          <Typography>
+            <Heading>About Mux</Heading>
+            <Paragraph>
+              This app connects to Mux and allows you to upload videos to your
+              content in Contentful. After entering your API Credentials then
+              choose which JSON fields in your content model you would like to
+              configure for Mux Video. For those configured fields you'll get a
+              video uploader in the Contentful UI. Your videos will be
+              transcoded, stored and delivered by{' '}
+              <a href="https://mux.com">Mux</a>.
```

Comment:
Let's use Forma-36's `<TextLink>` as well as get this to open in a new page with some security enabled.
```suggestion
              <TextLink href="https://mux.com" rel="noopener noreferrer" target="_blank">Mux</TextLink>.
```